applications/ConvectionDiffusionApplication/custom_elements/symbolic_d_convection_diffusion_explicit.cpp

```diff
@@ -0,0 +1,1783 @@
+{
+    KRATOS_TRY;
+
+    BaseType::Initialize(rCurrentProcessInfo);
+    // Resize and intialize output
+    if (mUnknownSubScale.size() != TNumNodes) // number integration points = number nodes
+        mUnknownSubScale.resize(TNumNodes, false);
+    mUnknownSubScale = ZeroVector(TNumNodes);
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template< unsigned int TDim, unsigned int TNumNodes >
+void SymbolicDConvectionDiffusionExplicit<TDim,TNumNodes>::FinalizeSolutionStep(
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    BaseType::FinalizeSolutionStep(rCurrentProcessInfo);
+    // Element variables
+    ElementVariables rVariables;
+    this->InitializeEulerianElement(rVariables,rCurrentProcessInfo);
+    // Reading integration points and local gradients
+    const auto& r_geometry = this->GetGeometry();
+    const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints(this->GetIntegrationMethod());
+
+    // Iterate over integration points to update subscales on the gauss point
+    for (unsigned int g = 0; g < integration_points.size(); g++) {
+        // Caluclate N on the gauss point "g"
+        rVariables.N = row(rVariables.N_gausspoint,g);
+        // Compute tau
+        this->CalculateTau(rVariables);
+        // Retrieve unknown belonging to subgrid scale space on gauss integration point g
+        rVariables.unknown_subscale = mUnknownSubScale(g);
+        // Update unknown belonging to subgrid scale space on gauss integration point g
+        this->UpdateUnknownSubgridScaleGaussPoint(rVariables,g);
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template< unsigned int TDim, unsigned int TNumNodes >
+void SymbolicDConvectionDiffusionExplicit<TDim,TNumNodes>::Calculate(
+    const Variable<double>& rVariable,
+    double& Output,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    auto& r_geometry = this->GetGeometry();
+    const unsigned int local_size = r_geometry.size();
+    VectorType rhs_oss;
+    this->CalculateOrthogonalSubgridScaleSystem(rhs_oss,rCurrentProcessInfo);
+    for (unsigned int i_node = 0; i_node < local_size; i_node++) {
+        #pragma omp atomic
+        r_geometry[i_node].GetValue(rVariable) += rhs_oss[i_node];
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template< unsigned int TDim, unsigned int TNumNodes >
+void SymbolicDConvectionDiffusionExplicit<TDim,TNumNodes>::CalculateOrthogonalSubgridScaleSystem(
```

Comment:
The name of the method is misleading. You are not calculating the system but the RHS.

Reply:
Done. I called it `CalculateOrthogonalSubgridScaleRHSInternal.` If you agree with the naming, please close the conversation.